### PR TITLE
DOC-8367 added language for immutable backups

### PR DESCRIPTION
**Pull request by ashleykwtam**
Addresses DOC-8367:
- Currently our docs have no language around immutable backups and we only refer to this through object locking. I've added this for search visibility

### CODE REVIEW

File: src/current/v22.2/backup.md [hunk 1]
```diff
@@ -42,7 +42,7 @@ To view the contents of an backup created with the `BACKUP` statement, use [`SHO
 ### Storage considerations
 
 - [HTTP storage](use-a-local-file-server.html) is not supported for `BACKUP` and `RESTORE`.
-- Modifying backup files in the storage location could invalidate a backup, and therefore, prevent a restore. In v22.1 and later, **we recommend enabling [object locking](use-cloud-storage.html#object-locking) in your cloud storage bucket.**
+- Modifying backup files in the storage location could invalidate a backup, and therefore, prevent a restore. In v22.1 and later, **we recommend enabling [object locking](use-cloud-storage.html#immutable-storage) in your cloud storage bucket.**
```
Comment (MichaelTrestman) [inline src/current/v22.2/backup.md:45]:
```suggestion
- Modifying backup files in the storage location could invalidate a backup, and therefore, prevent a restore. In v22.1 and later, we recommend enabling [object locking](use-cloud-storage.html#immutable-storage) in your cloud storage bucket.
```

File: src/current/v22.2/use-cloud-storage.md [hunk 2]
```diff
@@ -227,9 +227,9 @@ For guidance on adding a user to a bucket's policy, see [Add a principal to a bu
 
 ## Additional cloud storage feature support
 
-### Object locking
+### Immutable storage
 
-To complete a backup successfully, `BACKUP` requires [read and write permissions](backup.html#required-privileges) to cloud storage buckets. Delete and overwrite permissions are **not** required. As a result, you can write backups to cloud storage buckets with object locking enabled. This allows you to store backup data using a _write-once-read-many (WORM)_ model, which refers to storage that prevents any kind of deletion or modification to the objects once written.
+To complete a backup successfully, `BACKUP` requires [read and write permissions](backup.html#required-privileges) to cloud storage buckets. Delete and overwrite permissions are **not** required. As a result, you can write backups to cloud storage buckets with object locking enabled. This allows you to store backup data using a _write-once-read-many (WORM)_ model, which refers to storage that prevents any kind of deletion, encryption or modification to the objects once written.
```
Comment (MichaelTrestman) [inline src/current/v22.2/use-cloud-storage.md:232]: I think this is a more precise restatement, if I understand what this means?

```suggestion
To execute `BACKUP` statements, a SQL user must have [read and write permissions](backup.html#required-privileges) to cloud storage buckets. Delete and overwrite permissions are **not** required. As a result, you can write backups to cloud storage buckets with object locking enabled. This allows you to store backup data using a _write-once-read-many (WORM)_ model, which refers to storage that prevents any kind of deletion, encryption or modification to the objects once written.
```

File: src/current/v23.1/backup-and-restore-overview.md [hunk 3]
```diff
@@ -79,8 +79,8 @@ We recommend taking backups to [cloud storage](use-cloud-storage.html) and enabl
 
 For detail on additional cloud storage features CockroachDB supports:
 
-- [Object locking](use-cloud-storage.html#object-locking) to prevent backups from being overwritten or deleted.
-- [Storage Class (AWS S3 only)](use-cloud-storage.html#amazon-s3-storage-classes) to set a specific storage class for your backups.
+- Prevent backups from being overwritten or deleted with [immutable storage buckets](use-cloud-storage.html#immutable-storage).
```
Comment (MichaelTrestman) [inline src/current/v23.1/backup-and-restore-overview.md:82]:
```suggestion
- Preventing backups from being overwritten or deleted with [immutable storage buckets](use-cloud-storage.html#immutable-storage).
```

**Review by MichaelTrestman — APPROVED** [review]
LGTM, just that one suggested clarification.

**Review by livlobo — APPROVED** [review]
lgtm. Assuming this will show up in the search?
Also, I didn't see it in the diff, but I'm assuming we're chaning the title fom --> Object locking to "Immutable storage"

**Comment by ashleykwtam** [comment]
> lgtm. Assuming this will show up in the search?

Yes it should. I'll confirm upon merge

**Comment by ashleykwtam** [comment]
> Also, I didn't see it in the diff, but I'm assuming we're chaning the title fom --> Object locking to "Immutable storage"

Yes! title change is here - https://deploy-preview-17495--cockroachdb-docs.netlify.app/docs/stable/use-cloud-storage#immutable-storage

**Pull request merged**